search-service/src/main/kotlin/com/egm/stellio/search/listener/ObservationEventListener.kt

```diff
@@ -99,18 +101,20 @@ class ObservationEventListener(
         }
     }
 
-    suspend fun handleAttributeAppendEvent(observationEvent: AttributeAppendEvent): Either<APIException, Unit> {
-        val expandedPayload = expandAttribute(
+    suspend fun handleAttributeAppendEvent(
+        observationEvent: AttributeAppendEvent
+    ): Either<APIException, Unit> = either {
+        val expandedAttribute = expandAttribute(
             observationEvent.attributeName,
             observationEvent.operationPayload,
             observationEvent.contexts
         )
 
-        val ngsiLdAttributes = parseToNgsiLdAttributes(expandedPayload)
-        return entityPayloadService.appendAttributes(
+        val ngsiLdAttribute = expandedAttribute.toNgsiLdAttribute().bind()
+        entityPayloadService.appendAttributes(
             observationEvent.entityId,
-            ngsiLdAttributes,
-            expandedPayload,
+            listOf(ngsiLdAttribute),
+            expandedAttribute.toExpandedAttributes(),
```

Comment:
Maybe use a variable to expandAttributes to avoid having to do expansion two times ?